src/api/include/migraphx/migraphx.h

```diff
@@ -205,7 +205,7 @@ migraphx_status migraphx_dynamic_dimensions_assign_to(migraphx_dynamic_dimension
 
 migraphx_status
 migraphx_dynamic_dimensions_create(migraphx_dynamic_dimensions_t* dynamic_dimensions,
-                                   const_migraphx_dynamic_dimension_t* ptr,
+                                   const const_migraphx_dynamic_dimension_t* ptr,
```

Comment:
`const const_migraphx*` ? Why is that ?

Reply:
First const is for the handle, and the second const is for what the handle points to.

This is using an extra pointer here because it takes an array of `const_migraphx_dynamic_dimension_t` which is passed by pointer. Each `const_migraphx_dynamic_dimension_t` is a pointer as well that points to data that we cant/dont modify. So we are taking a pointer to an array of pointers. Then the additional `const` is because we dont modify the pointers in the array.

Reply:
Was cppcheck complaning about this pointer not being `const` ? because we also have some other `create()` methods which would also require `const` pointer to `const` object in that case.

Reply:
If i expand this, it would become `const const migraphx_dynamic_dimension*` . Should it be instead `const _migraphx_dynamic_dimension_t* const ptr` instead?

Reply:
> Should it be instead const _migraphx_dynamic_dimension_t* const ptr instead?

The variable itself is not check for const and this might produce warnings from the compiler when making function parameters const.